nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/GFFileHandler.java

```diff
@@ -601,11 +585,10 @@ void initializePump() {
                         }
                     }
                 }
-            );
-        } else {
-            drainAllPendingRecords();
-            flush();
-        }
+        };
+        pump.setName("GFFileHandler log pump");
+        pump.setDaemon(true);
+        pump.start();
```

Comment:
If the pending records are not drained and flushed they will remain unwritten to the log if the logToFile is set to false during operation.